cupyx/scipy/sparse/compressed.py

```diff
@@ -558,9 +558,9 @@ def _major_slice(self, idx, copy=False):
         M = len(range(start, stop, step))
         new_shape = self._swap(M, N)
 
-        if M == 0 or self.nnz == 0:
-            return self.__class__(new_shape)
         if step == 1:
+            if M == 0 or self.nnz == 0:
+                return self.__class__(new_shape, dtype=self.dtype)
```

Comment:
Could you explain why the shortcut requires `step == 1`?

Reply:
I fixed it for the dtype of the return value. SciPy's sparse array returns a sparse matrix array of `self.dtype` type if `step == 1`, but returns float64 otherwise...